torch/csrc/api/include/torch/nn/options/conv.h

```diff
@@ -11,27 +12,31 @@ namespace nn {
 /// Options for a `D`-dimensional convolution module.
 template <size_t D>
 struct ConvOptions {
+  typedef c10::variant<enumtype::kZeros, enumtype::kCircular> padding_mode_t;
+
+  ConvOptions() {}
+
   ConvOptions(
-      int64_t input_channels,
-      int64_t output_channels,
+      int64_t in_channels,
+      int64_t out_channels,
       ExpandingArray<D> kernel_size) :
-                input_channels_(input_channels),
-                output_channels_(output_channels),
+                in_channels_(in_channels),
+                out_channels_(out_channels),
                 kernel_size_(std::move(kernel_size)) {}
 
   /// The number of channels the input volumes will have.
   /// Changing this parameter after construction __has no effect__.
-  TORCH_ARG(int64_t, input_channels);
+  TORCH_ARG(c10::optional<int64_t>, in_channels) = c10::nullopt;
 
   /// The number of output channels the convolution should produce.
   /// Changing this parameter after construction __has no effect__.
-  TORCH_ARG(int64_t, output_channels);
+  TORCH_ARG(c10::optional<int64_t>, out_channels) = c10::nullopt;
 
   /// The kernel size to use.
   /// For a `D`-dim convolution, must be a single number or a list of `D`
   /// numbers.
   /// This parameter __can__ be changed after construction.
-  TORCH_ARG(ExpandingArray<D>, kernel_size);
+  TORCH_ARG(c10::optional<ExpandingArray<D>>, kernel_size) = c10::nullopt;
```

Comment:
I need to make these three arguments optional (and check that they are defined in `reset`), because the options for `F::conv{1,2,3}d` doesn't need these three arguments.